src/librustc_middle/ty/query/keys.rs

```diff
@@ -25,6 +25,18 @@ pub trait Key {
     fn default_span(&self, tcx: TyCtxt<'_>) -> Span;
 }
 
+impl Key for () {
```

Comment:
Wait, this is new? I thought there was precedent for this. I guess we can still go ahead with this.

Reply:
I checked, no query have `()` as argument/key type yet, i think.